### DISCUSSION TITLE
Auto Deploy from GitHub

### DISCUSSION THREAD

**Discussion by axmc1** [deta/cloud-docs#256] [Ideas] [answered]
Heyo,

I think the title explains most of it but I'll go into more detail.

Auto Deployment is always a nice QOL feature to have and I think it could become very useful in Deta Micros too.
My suggestion would be to add an auto-deploy feature, that "auto-deploys" your app when you commit to the main branch in a repo. If committed to a repo that is not the main branch, it'll deploy that branch to a preview domain to allow testing on Deta micros (behavior). That connection would always be an optional feature and could later be worked on to support GitLab and other version control software.

**Comment by cofob** [comment, marked as answer]
The Github/gitlab CI already supports this. Just make a separate workflow in which you install the Deta CLI, add tokens via secrets and call `deta deploy`.

**Reply by axmc1** [reply]
Didn't quite think it this way, but yea that should work. Thx!

**Comment by abdelhai** [comment]
there's also this https://github.com/marketplace/actions/deploy-to-deta
